### PR TITLE
Added base implementation of neumorphism sample

### PR DESCRIPTION
**Pull request by iKrupp**

### CODE REVIEW

File: lib/crypto_wallet/neu/utils/button_animator.dart [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    super.didUpdateWidget(old);
+  }
+
+  @override
+  void dispose() {
+    _animationController.dispose();
+    super.dispose();
+  }
+
+  bool _buttonHeldDown = false;
```
Comment (anemchinova) [inline lib/crypto_wallet/neu/utils/button_animator.dart:45]: do you want to move it to the top of the class?

File: lib/crypto_wallet/neu/utils/button_animator.dart [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  @override
+  void initState() {
+    super.initState();
+    _animationController = widget.animationController;
+  }
+
+  @override
+  void didUpdateWidget(ButtonAnimator old) {
+    super.didUpdateWidget(old);
+  }
```
Comment (anemchinova) [inline lib/crypto_wallet/neu/utils/button_animator.dart:37]: it does nothing

File: lib/crypto_wallet/neu/utils/button_animator.dart [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      _animate();
+    }
+  }
+
+  void _animate() {
+    if (_animationController.isAnimating) return;
+    final bool wasHeldDown = _buttonHeldDown;
+    final TickerFuture ticker = _buttonHeldDown
+        ? _animationController.animateTo(1.0, duration: kOutDuration)
+        : _animationController.animateTo(0.0, duration: kInDuration);
```
Comment (anemchinova) [inline lib/crypto_wallet/neu/utils/button_animator.dart:73]: why not forward / reverse?

File: lib/crypto_wallet/neu/utils/button_animator.dart [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    ticker.then<void>((void value) {
+      if (mounted && wasHeldDown != _buttonHeldDown) {
+        _animate();
+      }
+    });
```
Comment (anemchinova) [inline lib/crypto_wallet/neu/utils/button_animator.dart:78]: nit: await ticker ?

**Review by anemchinova — APPROVED** [review]
a couple of nits, looks good overall!

File: lib/crypto_wallet/neu/utils/button_animator.dart (hunk 3 above)
Comment (iKrupp) [reply lib/crypto_wallet/neu/utils/button_animator.dart:73]: These methods don't allow to set the duration

Comment (anemchinova) [reply lib/crypto_wallet/neu/utils/button_animator.dart:68]: you can set different duration for forward and reverse animation

**Pull request merged**